app/routes/api.js

```diff
@@ -3,37 +3,69 @@
 var express = require('express');
 var router = express.Router();
 
+var Account = require('../models/account');
 var Log = require('../models/log');
 
-// logs ======================================================================
-router.post('/logs', function(req, res) {
-    var userId = req.body.userId;
+module.exports = function(passport) {
+    // logs ======================================================================
+    router.post('/logs/', passport.authenticate('bearer', { session: false }), function(req, res) {
+        // force application/json content-type
+        if(req.headers["content-type"] !== 'application/json') {
+            return res.status(400).send('Error: content-type must be application/json');
+        }
 
-    req.body.logs.forEach(function(log) {
-        var newLog = new Log();
+        if(!req.body.logs) {
+            return res.status(400).send('Error: invalid body content');
+        }
 
-        newLog.userId = userId;
-        newLog.type = log.type;
-        newLog.data = JSON.stringify(log.data);
-
-        newLog.save(function(err) {
-            if (err) {
-                res.status(500).send(err);
+        Account.findOne({ 'key': req.user.key }, function(err, account) {
+            if(err) {
+                return res.status(500).send(err);
             }
+
+            req.body.logs.forEach(function(log) {
+                var newLog = new Log({
+                    type: log.type,
+                    data: JSON.stringify(log.data),
+                    account: account._id
+                });
+
+                if(!newLog.type || !newLog.data) {
+                    return res.status(400).send('Error: invalid body content');
+                }
+
+                newLog.save(function(err) {
+                    if (err) {
+                        return res.status(500).send(err);
+                    }
+
+                    account.logs.push(newLog);
+
+                    account.save(function(err) {
+                        if (err) {
+                            return handleError(err);
+                        }
+
+                        res.send(200);
+                    });
+                });
+            });
         });
     });
 
-    res.send(200);
-});
+    router.get('/logs', passport.authenticate('bearer', { session: false }), function(req, res) {
+        Account.findOne({ 'key': req.user.key }).populate('logs').exec(function(err, account) {
+            if(err) {
+                return handleError(err);
+            }
 
-router.get('/logs', function(req, res) {
-    Log.find({ 'userId': req.query.userId }, function (err, logs) {
-        res.send(logs);
+            res.send(account.logs);
+        });
     });
-});
 
-router.get('/', function(req, res) {
-    res.send('Welcome to our API!');
-});
+    router.get('/', passport.authenticate('bearer', { session: false }), function(req, res) {
```

Comment:
Oh for sure, it will be updated. It's just been there as a way to quickly and easily make sure all is working properly. It will be killed in the next PR I'm sure.